src/App/Pages/Accounts/LoginPasswordlessRequestViewModel.cs

```diff
@@ -0,0 +1,212 @@
+            {
+                return;
+            }
+
+            try
+            {
+                var response = await _authService.GetPasswordlessLoginResponse(_requestId, _requestAccessCode);
+
+                if (!response.RequestApproved)
+                {
+                    return;
+                }
+
+                StopCheckLoginRequestStatus();
+
+                var decKey = await _cryptoFunctionService.RsaDecryptAsync(response.Key, _requestKeyPair.Item2);
+                var decPasswordHash = await _cryptoFunctionService.RsaDecryptAsync(response.MasterPasswordHash, _requestKeyPair.Item2);
+
+                var authResult = await _authService.LogInPasswordlessAsync(Email, _requestAccessCode, _requestId, decKey, decPasswordHash);
+                await AppHelpers.ResetInvalidUnlockAttemptsAsync();
+
+                if (authResult.CaptchaNeeded)
+                {
+                    if (await HandleCaptchaAsync(authResult.CaptchaSiteKey))
+                    {
+                        await CheckLoginRequestStatus();
+                        _captchaToken = null;
+                    }
+                    return;
+                }
+                _captchaToken = null;
+
+                await _deviceActionService.HideLoadingAsync();
+
+                if (authResult.TwoFactor)
+                {
+                    StartTwoFactorAction?.Invoke();
+                }
+                else if (authResult.ForcePasswordReset)
+                {
+                    UpdateTempPasswordAction?.Invoke();
+                }
+                else
+                {
+                    var task = Task.Run(async () => await _syncService.FullSyncAsync(true));
+                    LogInSuccessAction?.Invoke();
+                }
+            }
+            catch (Exception ex)
+            {
+                StartCheckLoginRequestStatus();
+                HandleException(ex);
+            }
+        }
+
+        private async Task CreatePasswordlessLoginAsync()
+        {
+            await _deviceActionService.ShowLoadingAsync(AppResources.Loading);
+
+            var keyPair = await _cryptoFunctionService.RsaGenerateKeyPairAsync(2048);
+            var generatedFingerprintPhrase = await _cryptoService.GetFingerprintAsync(Email, keyPair.Item1);
+            var fingerprintPhrase = string.Join("-", generatedFingerprintPhrase);
+            var publicB64 = Convert.ToBase64String(keyPair.Item1);
+            var accessCode = await _passwordGenerationService.GeneratePasswordAsync(new PasswordGenerationOptions(true) { Length = 25 });
+            var response = await _authService.PasswordlessCreateLoginRequestAsync(_email, publicB64, accessCode, fingerprintPhrase);
+
+            if (response != null)
+            {
+                FingerprintPhrase = fingerprintPhrase;
+                _requestAccessCode = accessCode;
+                _requestId = response.Id;
+                _requestKeyPair = keyPair;
+            }
```

Comment:
If possible, could you move this to a service that makes all the operations and returns an struct with the things needed here: `fingerprintPhrase, accessCode, response.Id, keyPair`?
Is the `Length = 25` something fixed and we can move to a const somewhere or will it change to some setting?
I believe there's already a const for the fingerprint phrase separator